staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -227,10 +199,34 @@ func (s *store) conditionalDelete(ctx context.Context, key string, out runtime.O
 	}
 	for {
 		origState, err := s.getState(getResp, key, v, false)
-		if err != nil {
+		if err != nil && !storage.IsTransformerError(err) {
 			return err
 		}
-		if err := preconditions.Check(key, origState.obj); err != nil {
+		if err != nil && storage.IsTransformerError(err) && preconditions != nil {
+			return err
+		}
+		if err != nil && storage.IsTransformerError(err) && preconditions == nil {
+			// in this rare case, we skip checking
+			// validateDeletion, to give user a way to delete the
+			// key via the apiserver in case of data corruption.
+			startTime := time.Now()
+			// We don't check the transaction response as we know
+			// the transformer cannot interpret it anyway.
+			_, txnErr := s.client.KV.Txn(ctx).If().Then(
```

Comment:
should still check `clientv3.Compare(clientv3.ModRevision(key), "=", origState.rev)` to make sure we're deleting the same revision we couldn't interpret

Reply:
don't we need txnResp to check `txnResp.Succeeded`, even if we don't decode?

Reply:
Good point. See the last commit. Note that I had to modify `getState()` as well, because otherwise origState.rev would be nil.